src/js/qty-input.ts

```diff
@@ -22,24 +22,191 @@
  * @copyright Since 2007 PrestaShop SA and Contributors
  * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
  */
+import { Toast } from 'bootstrap';
+import SelectorsMap, { listing } from './selectors-map';
 
-export default function initQuantityInput(selector: String) {
-  $(document).ready(() => {
-    $(selector).inputSpinner({
-      decrementButton: '<i class="material-icons">expand_more</i>',
-      incrementButton: '<i class="material-icons">expand_less</i>',
-      buttonsClass: '',
-      buttonsWidth: '1.25rem',
-      /* eslint-disable */
-      template: 
-          '<div class="input-group ${groupClass}">' +
-            '<input type="text" inputmode="decimal" style="text-align: ${textAlign}" class="form-control form-control-text-input"/>' +
-            '<div class="qty-right">' +
-              '<button style="min-width: ${buttonsWidth}" class="btn btn-increment ${buttonsClass} btn-plus" type="button">${incrementButton}</button>' +
-              '<button style="min-width: ${buttonsWidth}" class="btn btn-decrement ${buttonsClass} btn-minus" type="button">${decrementButton}</button>' +
-            '</div>' +
-          '</div>'
-      /* eslint-enable */
+export default function initQuantityInput(selector = SelectorsMap.qtyInput.default) {
+  const qtyInputNodeList = document.querySelectorAll(selector);
+  const decrementIcon = 'E15B';
+  const incrementIcon = 'E145';
+
+  if (qtyInputNodeList) {
+    qtyInputNodeList.forEach(function(qtyInput) {
+      const qtyInputWrapper = qtyInput.parentElement;
+
+      if (qtyInputWrapper && qtyInputWrapper.childElementCount === 1) {
+        const decrementButton = createSpinButton(decrementIcon);
+        decrementButton.addEventListener('click', () => changeQuantity(<HTMLInputElement>qtyInput, -1));
```

Comment:
I think we can find a way to avoid these cast, I'll fetch your branch and take a look later